kubernetes/deployment-staging.tmpl

```diff
@@ -0,0 +1,300 @@
+      app: talk-staging-sidekiq
+  template:
+    metadata:
+      labels:
+        app: talk-staging-sidekiq
+    spec:
+      containers:
+        - name: talk-staging-sidekiq
+          image: zooniverse/talk-api:__IMAGE_TAG__
+          resources:
+            requests:
+              memory: "700Mi"
+              cpu: "50m"
+              ephemeral-storage: "1Gi"
+            limits:
+              memory: "700Mi"
+              cpu: "500m"
+              ephemeral-storage: "1Gi"
+          livenessProbe:
+            exec:
+              command:
+                - /rails_app/docker/sidekiq_status
+            initialDelaySeconds: 20
+          args: ["/rails_app/scripts/docker/start-sidekiq.sh"]
+          envFrom:
+          - secretRef:
+              name: talk-common-env-vars
+          - secretRef:
+              name: talk-staging-env-vars
+          - configMapRef:
+              name: talk-staging-shared
+          volumeMounts:
+          - mountPath: /tmp
+            name: talk-staging-dumpworker-data
+      volumes:
+        - name: talk-staging-dumpworker-data
+          emptyDir: {}
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: talk-staging-redis
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: azurefile
+  resources:
+    requests:
+      storage: 1Gi
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: talk-staging-redis
+  labels:
+    app: talk-staging-redis
+spec:
+  replicas: 1
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: talk-staging-redis
+  template:
+    metadata:
+      labels:
+        app: talk-staging-redis
+    spec:
+      tolerations:
+      - key: "servicelife"
+        operator: "Equal"
+        value: "longlife"
+        effect: "NoSchedule"
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: servicelife
+                operator: In
+                values:
+                - longlife
+      initContainers:
+        - name: disable-thp
+          image: busybox
+          volumeMounts:
+            - name: host-sys
+              mountPath: /host-sys
+          command: ["sh", "-c", "echo never >/host-sys/kernel/mm/transparent_hugepage/enabled"]
+      containers:
+        - name: talk-staging-redis
+          image: redis:6.0.5
+          resources:
+            requests:
+              memory: "100Mi"
+              cpu: "10m"
+            limits:
+              memory: "100Mi"
+              cpu: "500m"
+          volumeMounts:
+          - name: talk-staging-redis-data
+            mountPath: "/data"
+      volumes:
+        - name: talk-staging-redis-data
+          persistentVolumeClaim:
+            claimName: talk-staging-redis
+        - name: host-sys
+          hostPath:
+            path: /sys
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: talk-staging-redis
```

Comment:
I think we can combine to the panoptes staging redis service and drop this nginx setup. Also it appears that `REDIS_URL` in `talk-staging-env-vars` is using panoptes redis not this one.

Reply:
Nice, good catch on the secret. Currently, Talk does in fact use Panoptes' redis server on a different db, so I copied it over. Now that it uses its own, I'll switch the secret to using `redis://talk-staging-redis:6379`, which should point it at the redis container, I think. Same for production. See above for why I'm keeping redis in this deployment, we could probably combine more than a couple while we're at it.

Reply:
My thinking is staging instances can be shared, production should probably isolate by default.  

I don't mind which way we go tbh, as long as we're not running too many redis services on 1 custom k8s instance node and we start seeing timeouts due to cpu starvation, especially in the critical services like panoptes, caesar, etc.